spacy/language.py

```diff
@@ -1615,6 +1615,15 @@ def from_config(
                 or lang_cls is not cls
             ):
                 raise ValueError(Errors.E943.format(value=type(lang_cls)))
+        # Warn about require_gpu if a jupyter notebook + cupy + mismatched
+        # contextvars vs. thread ops are detected
+        if is_in_jupyter():
```

Comment:
Not your fault but this kis kinda ugly so maybe we should move it into a helper function? 🤔

Reply:
Yeah I think the helper function is worth it, especially considering the `Language.from_config` is already not the easiest method to read/follow. 

It'll hopefully all be temporary anyways.

Reply:
The bug report is old enough that I wouldn't have much short-term hope about an actual solution.